packages/shared-components/src/components/onto-repository-selector/onto-repository-selector.tsx

```diff
@@ -0,0 +1,102 @@
+  styleUrl: 'onto-repository-selector.scss',
+  shadow: false,
+})
+/**
+ * A StencilJS component that renders a repository selector dropdown. This component fetches the list of repositories
+ * from the `RepositoryService` and displays them in a dropdown. It allows users to select a repository, and automatically
+ * updates the current repository when a selection is made.
+ *
+ * @example
+ * <onto-repository-selector></onto-repository-selector>
+ */
+export class OntoRepositorySelector {
+  private repositoryService: RepositoryService;
+  private repositories: RepositoriesList;
+  private readonly subscriptions: Subscription[] = [];
+
+  /**
+   * A list of dropdown items representing repositories, updated when repositories change.
+   */
+  @State() items: DropdownItem<Repository>[] = [];
+
+  /**
+   * The currently selected repository, which will be shown in the dropdown button.
+   */
+  @State() currentRepository: Repository;
+
+  constructor() {
+    this.repositoryService = ServiceProvider.get(RepositoryService);
+    this.repositoryService.loadRepositories();
+    this.subscriptions.push(this.subscribeToRepositoriesChanged());
+    this.subscriptions.push(this.subscribeToSelectedRepositoryChanged());
+  }
+
+  /**
+   * Cleans up the subscriptions when the component is removed from the DOM to avoid memory leaks.
+   * This method unsubscribes from both the repository change and selected repository change events.
+   */
+  disconnectedCallback(): void {
+    this.subscriptions.forEach((subscription) => subscription.unsubscribe());
+  }
+
+  render() {
+    return (
+      <Host>
+        <onto-dropdown
+          class='onto-repository-selector'
+          onValueChanged={(newRepository: any) => this.onRepositoryChanged(newRepository)}
+          dropdownButtonName={<onto-repository-selector-button repository={this.currentRepository}></onto-repository-selector-button>}
+          dropdownAlignment={DropdownItemAlignment.RIGHT}
+          items={this.items}>
+        </onto-dropdown>
+      </Host>
+    );
+  }
+
+  private subscribeToRepositoriesChanged(): Subscription {
+    return this.repositoryService.onRepositoriesChanged()
+        .subscribe((repositories: RepositoriesList) => {
+          this.repositories = repositories || new RepositoriesList();
+          this.items = this.getRepositoriesDropdownItems();
+        })
+  }
+
+  private subscribeToSelectedRepositoryChanged(): Subscription {
+    return this.repositoryService.onSelectedRepositoryChanged()
+        .subscribe((selectedRepository) => this.changeRepository(selectedRepository));
+  }
+
+  private changeRepository(newRepository: Repository): void {
+    this.currentRepository = newRepository;
+    this.items = this.getRepositoriesDropdownItems();
+  }
+
+  private getRepositoriesDropdownItems(): DropdownItem<Repository>[] {
+    this.repositories.sortByLocationAndId();
+    return this.repositories.repositories
+        .filter((repository) => {
```

Comment:
I don't like this. The filtering should stay in the model!

Reply:
moved to model